src/Compiler/Service/service.fs

```diff
@@ -289,7 +291,9 @@ type FSharpChecker
     member ic.MatchBraces(fileName, source: string, options: FSharpProjectOptions, ?userOpName: string) =
         let userOpName = defaultArg userOpName "Unknown"
         let parsingOptions, _ = ic.GetParsingOptionsFromProjectOptions(options)
+
         ic.MatchBraces(fileName, SourceText.ofString source, parsingOptions, userOpName)
+        |> Cancellable.UsingToken
```

Comment:
This looks too easy to miss in future updates/additions to the service layer.
Is there something we can do to prevent such mistakes?

Reply:
Enforce it with types, which means use a specialized CE internally again :)